skbio/stats/distance/tests/test_base.py

```diff
@@ -1505,6 +1505,23 @@ def test_from_iterable_validate_false_condensed(self):
         )
         self.assertEqual(res, exp)
 
+    def test_from_iterable_with_invalid_diagonal(self):
+        iterable = (x for x in range(4))
```

Comment:
This can be `iterable = tuple(range(4))` or simply `iterable = range(4)`. But that's trivial.